repose-aggregator/components/filters/ip-classification/src/main/scala/org/openrepose/filters/ipclassification/IPClassificationFilter.scala

```diff
@@ -0,0 +1,153 @@
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.filters.ipclassification
+
+import java.net.URL
+import java.util
+import java.util.concurrent.ConcurrentLinkedQueue
+import javax.inject.{Inject, Named}
+import javax.servlet._
+import javax.servlet.http.HttpServletRequest
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import edazdarevic.commons.net.CIDRUtils
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.commons.utils.servlet.http.ReadableHttpServletResponse
+import org.openrepose.core.filter.FilterConfigHelper
+import org.openrepose.core.filter.logic.{FilterAction, FilterDirector}
+import org.openrepose.core.filter.logic.common.AbstractFilterLogicHandler
+import org.openrepose.core.filter.logic.impl.{FilterDirectorImpl, FilterLogicHandlerDelegate}
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.filters.ipclassification.config.IpClassificationConfig
+
+@Named
+class IPClassificationFilter @Inject()(configurationService: ConfigurationService) extends AbstractFilterLogicHandler
+with Filter
+with LazyLogging
+with UpdateListener[IpClassificationConfig] {
+
+  private final val DEFAULT_CONFIG = "ip-classification.cfg.xml"
+  private var initialized = false
+  private var config: String = _
+
+  type CIDRTuple = (String, CIDRUtils)
+  private val cidrList = new ConcurrentLinkedQueue[CIDRTuple]()
+  private var groupHeaderName: String = _
+  private var groupHeaderQuality: Double = _
+  private var userHeaderName: String = _
+  private var userHeaderQuality: Double = _
+
+  override def init(filterConfig: FilterConfig): Unit = {
+    config = new FilterConfigHelper(filterConfig).getFilterConfig(DEFAULT_CONFIG)
+    logger.info(s"Initializing using config $config")
+    val xsdURL: URL = getClass.getResource("/META-INF/config/schema/ip-classification.xsd")
+
+    configurationService.subscribeTo(
+      filterConfig.getFilterName,
+      config,
+      xsdURL,
+      this,
+      classOf[IpClassificationConfig]
+    )
+  }
+
+  override def destroy(): Unit = configurationService.unsubscribeFrom(config, this.asInstanceOf[UpdateListener[_]])
+
+  override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain): Unit = {
+    new FilterLogicHandlerDelegate(servletRequest, servletResponse, filterChain).doFilter(this)
+  }
+
+  override def handleRequest(httpServletRequest: HttpServletRequest, httpServletResponse: ReadableHttpServletResponse): FilterDirector = {
+    logger.debug("Handling HTTP request!")
+    val filterDirector = new FilterDirectorImpl()
+
+    getClassificationLabel(httpServletRequest.getRemoteAddr).foreach { label =>
+      filterDirector.requestHeaderManager.appendHeader(groupHeaderName, label, groupHeaderQuality)
+    }
+
+    //Always set the user header name to the current IP address
+    //TODO: apparently we append users as well?
```

Comment:
The old ip-identity filter did appending, because there's a quality associated with the user field. I suppose multiple filters could assign "users" based on whatever. It didn't make a lot of sense for me either, but I wasn't sure that I wanted to re-define that behavior.